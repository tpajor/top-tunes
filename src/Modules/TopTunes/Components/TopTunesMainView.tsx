import * as React from 'react';
import { LocationDescriptor } from 'history';

import TopTunesListContainer from '../Containers/TopTunesListContainer';

interface IProps {
  location: LocationDescriptor;
}
class TopTunesMainView extends React.PureComponent<IProps> {
  public render() {
    return (
      <div className='container container--page'>
        <h1 className='text text--title'>
          Welcome to top 100 iTunes list
        </h1>
        <TopTunesListContainer location={this.props.location}/>
      </div>
    );
  }
}

export default TopTunesMainView;
