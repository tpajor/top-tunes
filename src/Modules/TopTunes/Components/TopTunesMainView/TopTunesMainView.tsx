import * as React from 'react';

import TopTunesList from '../TopTunesList/TopTunesList';

class TopTunesMainView extends React.PureComponent {

  render() {
    return (
      <div className='container container--page'>
        <h1 className='text text--page-title'>
          Welcome to top 100 iTunes list
        </h1>
        <TopTunesList />
      </div>
    );
  }
}

export default TopTunesMainView;
