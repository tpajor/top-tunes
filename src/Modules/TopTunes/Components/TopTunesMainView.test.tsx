import * as React from 'react';
import { shallow } from 'enzyme';

import TopTunesMainView, { IProps } from './TopTunesMainView';

function setup() {
  const props: IProps = {
    location: 'test',
  };

  const wrapper = shallow(<TopTunesMainView {...props} />);

  return {
    wrapper,
    props,
  };
}

describe('TopTunesMainView component', () => {
  it('should match snapshot if no change was intended', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
