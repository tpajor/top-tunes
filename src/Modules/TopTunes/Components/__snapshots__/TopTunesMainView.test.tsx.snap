// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopTunesMainView component should match snapshot if no change was intended 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopTunesMainView
    location="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="text text--title"
        >
          Welcome to top 100 iTunes list
        </h1>,
        <Connect(TopTunesList)
          location="test"
        />,
      ],
      "className": "container container--page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome to top 100 iTunes list",
          "className": "text text--title",
        },
        "ref": null,
        "rendered": "Welcome to top 100 iTunes list",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "location": "test",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="text text--title"
          >
            Welcome to top 100 iTunes list
          </h1>,
          <Connect(TopTunesList)
            location="test"
          />,
        ],
        "className": "container container--page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to top 100 iTunes list",
            "className": "text text--title",
          },
          "ref": null,
          "rendered": "Welcome to top 100 iTunes list",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "location": "test",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
